config.test.json

```diff
@@ -7,7 +7,10 @@
 	},
 	"processor": {
 		"storage_dir":"/tmp/",
-		"user_agent": "Downloader v1",
+		"request_headers": {
+			"Accept": "*/*",
+			"User-Agent": "Downloader v1"
+		},
```

Comment:
Personally I believe that we could continue passing the `User-Agent` header separately, and just use the `request_headers` hash to pass any arbitrary headers to the server. Various libraries expose this header separately and it looks quite handy to me to use it that way.

However, this is only a suggestion.

Reply:
@dblia right! So, `curl` and `wget` are two of these libraries that expose a separate flag for the user agent alone. So this makes sense to keep as you suggest. Mind the following though, both of these libraries, when both the separate `--user-agent` flag is passed and also a `--header` with the `User-Agent: <some-user-agent>` then both of these tools will override the --user-agent flag. So we need to also maintain this behavior. I am fine with your suggestion. @0xdiba @agis any thoughts on that?

Reply:
@dblia This suggestion applies to the client (ferto), not the server (downloader).

Reply:
I too have no strong opinions on this, both approaches sound fine to me.
Although if we decide to keep both, we should be explicit about which takes precedence if a client send both a user agent both through the explicit field and through the request_headers struct. The behaviour doesn't have to be the same with curl or wget, but it has to be explicit and documented.